<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>772</width>
    <height>453</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STM32 CAN Baud Rate Calculator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../ubuntu14.04-64bit/STM32_CANBaudRate/ico.qrc">
    <normaloff>:/calculator.ico</normaloff>:/calculator.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>APB CLK:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="PCLKLineEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>36</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>MHz</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Baud Rate:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="baudRateLineEdit">
            <property name="text">
             <string>1000</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Kbps</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Sample Point:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="samplePointComboBox">
            <item>
             <property name="text">
              <string>&gt;=40%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&gt;=50%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&gt;=60%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&gt;=70%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&gt;=80%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&gt;=90%</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Error:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="errorComboBox">
            <item>
             <property name="text">
              <string>&lt;=1%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=2%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=3%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=4%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=5%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=6%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=7%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=8%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=9%</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>&lt;=10%</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="calculateBaudRatePushButton">
          <property name="text">
           <string>Calculate</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::WinPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QTableWidget" name="baudRateTableWidget">
           <column>
            <property name="text">
             <string>BS1</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>BS2</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>BRP</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Sample Point</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Baud Rate</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Error</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame_2">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>140</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::WinPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QTextBrowser" name="textBrowser">
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;BaudRate = APBCLK/BRP*(1+BS1+BS2)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;SamplePoint = ((1+BS1)/(1+BS1+BS2))*100%&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;Sample Point Recommend:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;75% when BaudRate &amp;gt; 800K&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;80% when BaudRate &amp;gt; 500K&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'SimSun'; font-size:9pt; color:#005500;&quot;&gt;87.5% when BaudRate &amp;lt;= 500K&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>772</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../ubuntu14.04-64bit/STM32_CANBaudRate/ico.qrc"/>
 </resources>
 <connections/>
</ui>
